src/Xamarin.Android.Tools.AndroidSdk/OS.cs

```diff
@@ -20,6 +22,13 @@ static OS ()
 			if (IsWindows) {
 				ProgramFilesX86 = GetProgramFilesX86 ();
 			}
+
+			if (IsWindows)
+				NativeLibraryFormat = "{0}.dll";
+			if (IsMac)
+				NativeLibraryFormat = "lib{0}.dylib";
+			if (!IsWindows && !IsMac)
+				NativeLibraryFormat = "lib{0}.so";
```

Comment:
You might want to special-case Linux here as well. In theory the code can run on e.g. the `*BSD` family of Unix systems (other than `macOS`) and if you know you're running on Linux then you can check what distribution you're on and use that distribution's mechanisms to detect Java instances (e.g. on Debian + family you can use `/usr/sbin/update-java-alternatives -l` but you can't use it, say, on Manjaro)

Reply:
I can't special-case systems I'm not running. :-)

BSD contributions welcome.  In the meantime, there is support for `$JAVA_HOME` and `$PATH` already.